### PR TITLE
elu activation added

### PR DESCRIPTION
**Pull request by Suraj520**
Added support for elu activation across all backends !

### CODE REVIEW

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Hello and thanks for the contribution, please use native functions instead of where such as `tf.keras.activations.elu` in the backends that have it, also please write the unit tests for them and move the implementation to the experimental API

**Comment by Suraj520** [comment]
@MahmoudAshraf97 - Did that change. However, I don't know how to write unit tests, yet ! Will have to check in the next free slot, maybe !

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Hi, please remove the function mse_loss from this PR and the other changes in the other files that are not related to your `elu` implementation
also can you write the tests for the new activation?

the paddle is missing the supported dtypes handles, and try to extend support for unsupported dtypes using compositional solutions

**Comment by Suraj520** [comment]
> Hi, please remove the function mse_loss from this PR and the other changes in the other files that are not related to your `elu` implementation also can you write the tests for the new activation?
> 
> the paddle is missing the supported dtypes handles, and try to extend support for unsupported dtypes using compositional solutions

@MahmoudAshraf97  - Sorry, I was a bit occupied with a kaggle competition submission deadline. I have removed the mse_loss from this PR as it was mistakenly pushed to this. It's meant for some other PR. Kindly check that it's pushed hard to that commit. 

Also, I'm looking into the unit tests now. Shall update you. Haven't written it ever, So may go through its learning curve, IMO. Thanks for your patience and understanding. If you've any resources to help me write the unit tests that you can share, It'll be great !

**Comment by Suraj520** [comment]
@MahmoudAshraf97 - Fixed the conflicts. Kindly review

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Hello and sorry for the late review, please make sure that the arguments of the function are consistent across the backends and the framework

**Comment by Suraj520** [comment]
> please make sure that the arguments of the function are consistent across the backends and the framework

No worries @MahmoudAshraf97 : I've fixed them ! Kindly review :smile:

**Comment by Suraj520** [comment]
> please remove the inplace argument and sticj to the ivy convention in the implementation, I think the closest function signature will be `ivy.thresholded_relu`

Done, Kindly review !

File: ivy/functional/ivy/activations.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ def deserialize(
 ]
 
 
+
```
Comment (Dsantra92) [inline ivy/functional/ivy/activations.py:100]: Any reason this extra line was added?

**Review by Dsantra92 — CHANGES_REQUESTED** [review]
I have left some reviews. These are mostly minor issues that need looking into. There are some inconsistent spacing, would you mind checking if the pre-commit is installed on your local?

File: ivy/functional/backends/tensorflow/experimental/activations.py [hunk 2]
```diff
@@ -36,6 +36,16 @@ def thresholded_relu(
     threshold = tf.cast(threshold, x.dtype)
     return tf.cast(tf.where(x > threshold, x, 0), x.dtype)
 
+@with_unsupported_dtypes({"2.12.0 and below": ("complex", "bool")}, backend_versiondef elu(
```
[the remark this answers is not in the document]
Comment (Suraj520) [reply ivy/functional/backends/tensorflow/experimental/activations.py:39]: Thanks for pointing out the glitch !

File: ivy/functional/backends/jax/activations.py [hunk 3]
```diff
@@ -14,7 +14,9 @@
 from ivy.functional.backends.jax import JaxArray
 
 
+
 @with_unsupported_dtypes({"0.4.10 and below": ("complex",)}, backend_version)
```
[the remark this answers is not in the document]
Comment (Suraj520) [reply ivy/functional/backends/jax/activations.py:18]: Made it to be consistent with rest function definitions out there !

File: ivy/functional/ivy/activations.py (hunk 1 above)
Comment (Suraj520) [reply ivy/functional/ivy/activations.py:100]: I guess that this got carried forward as a glitch when I migrated the code to experimental.

**Comment by Suraj520** [comment]
> I have left some reviews. These are mostly minor issues that need looking into. There are some inconsistent spacing, would you mind checking if the pre-commit is installed on your local?

Kindly review !

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Please add the container and array instance methods

**Comment by Suraj520** [comment]
> Please add the container and array instance methods

Hi, Sorry for the delay ! Can you please briefly explain how and where ?
Thanks in advance !

**Comment by MahmoudAshraf97** [comment]
@Suraj520 , in `ivy/data_classes/array` and `ivy/data_classes/container`, check the similar functions for examples

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by Suraj520** [comment]
> @Suraj520 , in `ivy/data_classes/array` and `ivy/data_classes/container`, check the similar functions for examples

Sure @MahmoudAshraf97 , Give me some time and I'll try to fix it

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
I see that you added the array and container instance methods in a commit but the changes are not in the final PR can you recheck? also remove all the changes in files that are not related to the PR in stable activations API in some backends

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_activations.py [hunk 4]
```diff
@@ -73,6 +73,41 @@ def test_thresholded_relu(
     )
 
 
+# elu
+@handle_test(
+    fn_tree="elu",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        shape=st.shared(helpers.get_shape(), key="elu"),
```
[the remark this answers is not in the document]
Comment (Suraj520) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_activations.py]: Got you ! @MahmoudAshraf97 ! 
Fixed the errors, Hopefully it should pass the tests.
